plenoptic/synthesize/synthesis.py

```diff
@@ -1037,7 +1037,10 @@ def load(self, file_path, map_location='cpu',
                         raise Exception(f"Saved and initialized {k} are different! Initialized: {getattr(self, k)}"
                                         f", Saved: {tmp_dict[k]}, difference: {getattr(self, k) - tmp_dict[k]}")
                 except RuntimeError:
-                    raise Exception(f"Attribute {k} is a different shape in saved and initialized versions!"
+                    if getattr(self, k).device != tmp_dict[k].device:
+                        raise Exception(f"Attribute {k} are on different devices in saved and initialized versions!"
+                                        f" Initialized: {getattr(self, k).device}, Saved: {tmp_dict[k].device}")
```

Comment:
I don't think this should  actually be an error, should it? If someone saves on the GPU, then sets `map_location='cpu'` to load on the CPU, this will throw an Exception, which we don't want.

But in order to compare them, they'll need to be on the same device, so maybe copy the saved version to the initialized device?

Reply:
You're right. So I can change the code to be asserting `torch.allclose(getattr(self, k).to(tmp_dict[k].device), tmp_dict[k])`.
The resulting behavior would be tolerating initiated and loaded data to be on different devices, and eventually using the device of `tmp_dict`, which is determined by `map_location` when it is not `None`.

Reply:
I think your solution is correct, though I guess the question is whether we want everything to end up on the device of `tmp_dict[k]` or of `self`. Right now, I think the final device is determined by `map_location`, which is what we want, I think